core/dbt/contracts/graph/nodes.py

```diff
@@ -971,6 +986,17 @@ def test_node_type(self):
         return "generic"
 
 
+@dataclass
+class UnitTestNode(CompiledNode):
+    resource_type: NodeType = field(metadata={"restrict": [NodeType.Unit]})
+    attached_node: Optional[str] = None
+    overrides: Optional[UnparsedUnitTestOverrides] = None
+
+    @property
+    def test_node_type(self):
+        return "unit"
```

Comment:
I think we can get rid of this since unit tests have their own NodeType (not NodeType.Test)

Reply:
Removed.